src/App.js

```diff
@@ -1,23 +1,61 @@
-import logo from './logo.svg';
-import './App.css';
+import axios from 'axios';
+import Footer from "./components/Footer";
+import Nav from "./components/Nav";
+import { API_URL } from './constants/constants';
+import { useEffect, useState } from 'react';
+import Folder from './components/Folder';
+
+const INITIAL_USER_VALUE = {
+  email:'',
+  id:0,
+  name:'',
+  profileImageSource:''
+};
+const INITIAL_FOLDER_VALUE = {
+  links:[],
+  name:'',
+  owner: {
+    name:'',
+    profileImageSource:''
+  }
+};
 
 function App() {
+  const [user, setUser] = useState(INITIAL_USER_VALUE);
+  const [folder, setFolder] = useState(INITIAL_FOLDER_VALUE);
+
+  const getUser = async() => {
+    try{
+      const response = await axios.get(`${API_URL}/user`);
+      const user = await response.data;
+      setUser(user);
+    } catch(error) {
+      alert(error)
+      setUser(INITIAL_USER_VALUE);
+    }
+  }
+
+  const getData = async() => {
+    try{
+      const response = await axios.get(`${API_URL}/folder`);
+      const { folder } = await response.data;
+      setFolder(folder);
+    } catch(error) {
+      alert(error)
+      setFolder(INITIAL_FOLDER_VALUE);
+    }
+  }
+  
+  useEffect(() => {
+    getUser();
+    getData()
+  },[])
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <Nav profile={user}/>
+      <Folder owner={folder.owner} name={folder.name} links={folder.links}></Folder>
+      <Footer/>
```

Comment:
컴포넌트 잘 쪼개주셨네요 👍 

지금으로서는, user상태는 Nav만 알면되고, folder상태는 Folder만 알면되는 군요. 그렇다면 해당 컴포넌트 안에서 상태를 선언해서 사용해도 무방해보여요. 이를테면,  Folder 컴포넌트가 이름이 Folder인데, folder 정보를 부모컴포넌트에게 받아야하는 상황인거죠. 그럼 Folder가 아니라 FolderTemplate, FolderLayout 이 더 정확한 이름이겠죠. 이름에 맞게 스스로의 책임을 수행할 수 있는 컴포넌트들로 구성하는게 유지보수에 장점이 있습니다. 

리렌더링에도 영향이 있는데, 지금 구조에서는 user상태가 바뀔 때,  App이 리렌더링될거고, user랑 상관없는 Folder 컴포넌트도 리렌더링되겠죠

Reply:
children이 없는 경우에는 self closing tag로 하는 편이 의도를 더 잘 드러낼 수 있습니다.

```suggestion
      <Folder
        owner={folder.owner} props생략 />
```